odoo/tools/cache.py

```diff
@@ -162,26 +165,61 @@ def determine_key(self) -> None:
 def log_ormcache_stats(sig=None, frame=None):   # noqa: ARG001 (arguments are there for signals)
     """ Log statistics of ormcache usage by database, model, and method. """
     from odoo.modules.registry import Registry
-    cache_entries = {}
-    current_db = None
-    cache_stats = ['Caches stats:']
-    for (dbname, model, method), stat in sorted(STAT.items(), key=lambda k: (k[0][0] or '~', k[0][1], k[0][2].__name__)):
-        dbname_display = dbname or "<no_db>"
-        if current_db != dbname_display:
-            current_db = dbname_display
-            cache_stats.append(f"Database {dbname_display}")
-        if dbname:   # mainly for MockPool
-            if (dbname, stat.cache_name) not in cache_entries:
-                cache = Registry.registries.d[dbname]._Registry__caches[stat.cache_name]
-                cache_entries[dbname, stat.cache_name] = Counter(k[:2] for k in cache.d)
-            nb_entries = cache_entries[dbname, stat.cache_name][model, method]
-        else:
-            nb_entries = 0
-        cache_name = stat.cache_name.rjust(25)
-        cache_stats.append(
-            f"{cache_name}, {nb_entries:6d} entries, {stat.hit:6d} hit, {stat.miss:6d} miss, {stat.err:6d} err, {stat.gen_time:10.3f}s time, {stat.ratio:6.1f}% ratio for {model}.{method.__name__}"
-        )
-    _logger.info('\n'.join(cache_stats))
+    log_msgs = ['Caches stats:']
+    # {dbname: sz_entries_all}
+    db_size = defaultdict(lambda: 0)
+    # {dbname: {(model, method): [sz_entries_sum, sz_entries_max, nb_entries, stat]}}
+    cache_stats = defaultdict(lambda: defaultdict(lambda: [0, 0, 0, None]))
+    
+    column_info = (
+        f"{'Cache Name':>25},"
+        f"{'Entry':>7},"
+        f"{'Memory SUM [Bytes]':>20},"
+        f"{'Memory MAX [Bytes]':>20},"
+        f"{'Hit':>6},"
+        f"{'Miss':>6},"
+        f"{'Err':>6},"
+        f"{'Gen Time [s]':>14},"
+        f"{'Ratio':>7},"
+        f"{'Model.Method':>15}"
+    )
+
+    for dbname, registry in Registry.registries.items():
+        db_cache_stats = cache_stats[dbname]
+        sz_entries_all = 0
+        for cache in registry._Registry__caches.values():
+            for cache_key, cache_value in cache.d.items():
+                model_name, method_name = cache_key[:2]
+                stats = db_cache_stats[(model_name, method_name)]
```

Comment:
`method_name` or `method` :thinking: ? in the last loop you use `method.__name__`, then I suppose that it is a method, right ?